docs/okeps/okep-5494-ovn-kubernetes-mcp-server.md

```diff
@@ -0,0 +1,740 @@
+      everything about the Kubernetes API state of the feature and to know what
+      the ovnkube pods were doing through the logs they generate during that
+      window
+    * **ovnkube-trace** which executes ovn-trace and ovs ofproto trace and
+      detrace commands (this tool doesn't support all scenarios - it's not
+      maintained well)
+    * Future-tool - ovnkube CLI (need to ask NVIDIA what's the status here)
+        * A place for K8s state-database syncer should/could potentially exist
+* OVN Layer - RHEL - OVN databases are gathered as part of must-gather for
+  offline troubleshooting
+    * **ovn-nbctl** commands that are executed on the ovnkube node pods to
+      understand what OVN-Kubernetes created into northbound database via
+      libovsdbclient transactions
+    * **ovn-sbctl** commands that are executed on the ovnkube node pods to
+      understand what northd created into southbound database
+    * **ovn-trace and detrace** commands that are executed on the ovnkube node
+      pods to understand simulated packet flow tracing based on flows in
+      southbound database
+    * **ovn-appctl -t ovn-controller ct-zone-list** to list all the conntrack
+      zone to OVN construct mapping for better understanding how the conntrack
+      commit of the packets happened
+* OpenvSwitch Layer - RHEL - openvswitch database is usually gathered as part
+  of sos-report for offline troubleshooting
+    * **ovs-ofctl dump-flows** to debug specially the br-ex/breth0 openflows
+      that OVN-Kubernetes creates on the gateway of each node
+    * **ovs-appctl dpctl/dump-flows** to trace live packets run on a specific
+      node's ovs container (KIND) or on the node (OpenShift)
+    * **ovs-appctl ofproto/trace** and detrace to run an ovs trace of the
+      packet based on the openflows
+    * **ovs-appctl dpctl/dump-conntrack** to know all the conntrack zones used
+      for a specific connection
+    * **ovs-vsctl** commands to list interfaces and bridges
+    * **retis** to see packet drops in ovs
+* Netfilter/Kernel Layer - RHEL - this information is usually gathered as part
+  of sos-report for offline troubleshooting
+    * **ip util commands** like **ip r** or **ip a** or **ip rule list** for
+      debugging VRFs, BGP learnt routes or the routes OVN-Kubernetes creates
+      on the node or custom routes that customer's add
+    * **nft list ruleset** to understand what rules were created by
+      OVN-Kubernetes specially in routingViaHost=true gateway mode
+    * **iptables-save** to list all iptables (Given iptables is deprecated,
+      I think we can skip this tool though for now 50% of ovn-kubernetes is
+      still on IPT)
+    * **conntrack** -L or -E on the host itself
+* TCPDUMP - external open source tools, can't be used for offline
+  troubleshooting
+    * **tcpdump** is used for packet capture and analysis
+    * **pwru** is used to know the kernel drop reason
```

Comment:
NIT: Link to pwru as I don't think it is generally common knowledge tool.

Reply:
done